extensions/amp-analytics/0.1/requests.js

```diff
@@ -41,8 +41,10 @@ export class RequestHandler {
    * @param {!../../../src/preconnect.Preconnect} preconnect
    * @param {function(string, !JsonObject)} handler
    * @param {boolean} isSandbox
+   * @param {!Element} ampAnalyticsElement
    */
-  constructor(ampdoc, request, preconnect, handler, isSandbox) {
+  constructor(ampdoc, request, preconnect, handler, isSandbox,
```

Comment:
One way is to have element as the first param. 
`constructor(element, request, preconnect, handler, isSandbox)`. Then we can get ampdoc from `#element.getAmpDoc()`
The most optimal way is to let FIE has its own ampdoc impl

Reply:
Done